utils/concatAlbumDescription.ts

```diff
@@ -0,0 +1,82 @@
+import { Album } from "@/schemas/collections.schemas";
+
+export default function concatAlbumDescription(album: Album) {
+  if (album.formats.length === 1) {
+    return album.formats
+      .map((format) => {
+        const { name, text, descriptions } = format;
+        const qty = parseInt(format.qty);
+        if (text && descriptions) {
+          if (qty === 1) {
+            return text + " " + name + " " + descriptions?.join(" ");
+          }
+
+          if (qty > 1) {
+            return (
+              text +
+              " " +
+              name +
+              " " +
+              descriptions?.join(" ") +
+              " " +
+              `(x${qty})`
+            );
+          }
+        }
+        if (text) {
+          if (qty === 1) {
+            return text + " " + name;
+          }
+
+          if (qty > 1) {
+            return text + " " + name + " " + `(x${qty})`;
+          }
+        }
+        if (descriptions) {
+          if (qty === 1) {
+            return name + " " + descriptions?.join(" ");
+          }
+
+          if (qty > 1) {
+            return name + " " + descriptions?.join(" ") + " " + `(x${qty})`;
+          }
+        }
+        if (text === undefined && descriptions === undefined) {
+          if (qty === 1) {
+            return name;
+          }
+
+          if (qty > 1) {
+            return name + " " + `(x${qty})`;
+          }
+        }
+      })
+      .join(", ");
+  } else {
```

Comment:
probably can adjust the map above in a way where you wouldn't need this if statement.

Reply:
Wasn't sure what you meant by this. Let me know if you have any other hints